core/src/main/scala/pickling/Macros.scala

```diff
@@ -225,11 +229,16 @@ trait PicklerMacros extends Macro with PickleMacros with FastTypeTagMacros {
           else reflectively("picklee", fir)(fm => putField(q"$fm.get.asInstanceOf[${fir.tpe}]"))
         } else if (fir.javaSetter.isDefined) {
           List(putField(getField(fir)))
-        } else {
+        } else if (fir.isParam) {
           reflectivelyWithoutGetter("picklee", fir)(fvalue =>
             tryPutField(q"$fvalue.asInstanceOf[scala.util.Try[${fir.tpe}]]"))
+        } else {
+          Nil
         }
       })
+      if (cir.fields.nonEmpty && putFields.isEmpty) {
+        throw PicklingException("No fields are captured. You need a custom pickler to handle this.")
+      }
```

Comment:
this doesn't seem clear enough to me. It could be something like `"fields $fieldNames are needed to construct $typeName, but pickling's generate macro didn't find a getter for these fields, so you will have to write a custom pickler"` ? I'm not sure I understand exactly why this happens though - if we found the field, why can't we use reflection to get it? Anyway whatever the surrounding explanation it's important to include the field and type names.